lsm-tree.cabal

```diff
@@ -342,9 +342,12 @@ test-suite lsm-tree-test
   hs-source-dirs: test
   main-is:        Main.hs
   other-modules:
+    Database.LSMTree.Class
+    Database.LSMTree.Class.Common
     Database.LSMTree.Class.Monoidal
     Database.LSMTree.Class.Normal
```

Comment:
So we added the unified API in #469 to allow writing tests against it. Here we add a *third* version of `IsTable`, the IO model etc. for the the unified API. But why not unify the model/tests as well, resulting in a single IO model and single `IsTable` class? That would make the tests much leaner.

The only argument against it that I can think of is that we're not yet sure which API(s) we want to provide in the end. If we later remove the unified API again, we might have to re-duplicate the model/tests. :shrug:

Reply:
Sure, it would be nice if we could eventually deduplicate the test stuff too, but it's a lot more work, and not strictly necessary for the statemachine tests

Reply:
Okay, but would be nice to discuss and tackle soonish then.

Reply:
Working on it in #486